### PR TITLE
Update the S3/AWS client package #2246

### LINKED ISSUE #2246: S3/AWS client package is outdated
### Increasing Access

We want to keep our codebase up to date.

### Feature enhancement details

We are interacting with the S3 bucket  for asset uploading using a package `@auth0/s3` which was last updated 5 years ago.  In contrast, the official Amazon `@aws-sdk/client-s3` package was last updated 3 days ago.

The current package `@auth0/s3` internally uses an outdated version of the `aws-sdk`, which logs a warning when running the server app:
> (node:28348) NOTE: We are formalizing our plans to enter AWS SDK for JavaScript (v2) into maintenance mode in 2023.
>
> Please migrate your code to use AWS SDK for JavaScript (v3).
> For more information, check the migration guide at https://a.co/7PzMCcy

Advantages of migration:
- Can use `async`/`await` instead of callbacks
- Better documentation
- Code is more future-proof

### PR DESCRIPTION
**Pull request by Ankush263**
Fixes #2246

Changes:

- Use `@aws-sdk/client-s3` package
- Use async await syntax

I have verified that this pull request:

* [✅ ] has no linting errors (`npm run lint`)
* [✅ ] has no test errors (`npm run test`)
* [✅ ] is from a uniquely-named feature branch and is up to date with the  `develop` branch.
* [✅ ] is descriptively named and links to an issue number, i.e. `Fixes #123`

### CODE REVIEW

File: server/controllers/aws.controller.js [hunk 1]
```diff
@@ -39,34 +40,34 @@ export function getObjectKey(url) {
...
 
 export function deleteObjectFromS3(req, res) {
   const { objectKey, userId } = req.params;
-  let fullObjectKey;
-  if (userId) {
-    fullObjectKey = `${userId}/${objectKey}`;
-  } else {
-    fullObjectKey = objectKey;
-  }
+  const fullObjectKey = userId ? `${userId}/${objectKey}` : objectKey;
```
Comment (lindapaiste) [inline server/controllers/aws.controller.js:70]: So much cleaner with the ternary!

**Comment by lindapaiste** [comment]
Thanks for working on this!  At first glance it looks great but I do need to give it a more careful review.

**Comment by Ankush263** [comment]
sure, go ahead 🚀🚀

**Review by raclim — APPROVED** [review]
Thanks for your work on this! 

I ran this locally and also created a [Release build for this update](https://app-ted01bd.release.editor.p5js.org/), and they both felt good to me so far. I'm going to merge this in, but I'll also continue to manually test the [`develop` branch](https://app-p5-js-web-editor-develop-branch.release.editor.p5js.org/) for the next few days, so feel free to make any additional suggestions or comments.

**Pull request merged**